app/core/guardian.py

```diff
@@ -22,6 +23,7 @@
 def guardian_from_query(query_result: Any) -> Guardian:
     return Guardian(
         guardian_id=query_result["guardian_id"],
+        name=query_result["name"],
```

Comment:
This is the solution to the 404 problem.  The underlying issue was just that a required field wasn't getting set.